### ISSUE TITLE
Assert-M365DSCBlueprint - Blueprint Report has many empty lines & accuracy of report feels off.

### ISSUE THREAD

**Issue by andikrueger** [microsoft/Microsoft365DSC#1748]
#### Details of the scenario you tried and the problem that is occurring

I did some testing today and exported the attached configuration. I renamed the config to a m365 document and ran 
``` powershell
Assert-M365DSCBlueprint -BluePrintUrl "M365TenantConfig.ps1.M365" -OutputReportPath "Assert.html" -Credentials $Credential -Verbose
```
This is one example of the differences between the export and the assert:
![image](https://user-images.githubusercontent.com/15608729/153035137-b0f2605a-916f-4862-8061-434ab66d5a7b.png)

[M365TenantConfig.ps1.M365.txt](https://github.com/microsoft/Microsoft365DSC/files/8025790/M365TenantConfig.ps1.M365.txt)
[Assert.html.txt](https://github.com/microsoft/Microsoft365DSC/files/8025792/Assert.html.txt)

**Comment by NikCharlebois** [comment]
Out of curiosity, if you export the configuration of the tenant you wish to assess, then use New-M365DSCDeltaReport to compare the two instead of the Assert-M365DSCBlueprint cmdlet, do you get similar results?

**Comment by andikrueger** [comment]
Nothing happens as soon as I add the -IsBlueprintAssessment.

If I run it without IsBlueprintAssessment the output is comparable to the Assert-M365DSCBluePrint

![image](https://user-images.githubusercontent.com/15608729/153181148-00afccd8-ea09-49e9-86b5-b9b0a7b9d9db.png)

**Comment by lewismartinSSC** [comment]
I see the same thing on my report for Exchange. I notice that when the PS1 file has either an empty array or "" for the property value then it will output an empty line.
![MicrosoftTeams-image](https://user-images.githubusercontent.com/71234219/153530605-3e7d277e-cc12-4384-8e84-72b217c08d6a.png)

**Comment by andikrueger** [comment]
I kind of get the feeling #1755 is related to this issue.

#1755 is about having ADDApplications with the same key within an exported configuration:
I think, the report does only check for the key parameters to be identical. As there could be several elements within a configuration that have the same keys, there is an issue with these resources within the reporting too.

**Comment by andikrueger** [comment]
This issue is related to #1725 as the Blueprint functionality relies on DSCParser. As long as there are issues parsing the configuration the assert will fail.

**Comment by lewismartinSSC** [comment]
> This issue is related to #1725 as the Blueprint functionality relies on DSCParser. As long as there are issues parsing the configuration the assert will fail.

did some testing and not seeing this same issue, getting same output as above with only one version installed.

**Comment by schmarrap** [comment]
Looking at the thread, I ended up doing the following:

1. Uninstall-Module Microsoft365dsc
2. Close all PowerShell instances, made sure the folder "Microsoft365dsc" is deleted C:\Program Files\WindowsPowerShell\Modules
3. Start new Powershell V5 and install-Module Microsoft365dsc
4. Update-M365DSCDependencies -force
5. Tested the following: 

> New-M365DSCReportFromConfiguration -ConfigurationPath <PahtanfFIlename> -Type JSON -OutputPath <ReportFilename and Path>

> New-M365DSCReportFromConfiguration -ConfigurationPath <PahtanfFIlename> -Type HTML -OutputPath <ReportFilename and Path>

Couldn't test excel yet
